web/modules/custom/collection/src/Plugin/Block/JoinCollectionBlock.php

```diff
@@ -56,12 +57,18 @@ class JoinCollectionBlock extends BlockBase implements ContainerFactoryPluginInt
    * @param \Drupal\Core\Session\AccountProxyInterface $user
    *   The current user.
    */
-  public function __construct(array $configuration, $plugin_id, $plugin_definition, RouteMatchInterface $current_route_match, AccountProxyInterface $user) {
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, RouteMatchInterface $current_route_match, AccountProxyInterface $user, ContextProviderInterface $collection_context) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
     $this->currentRouteMatch = $current_route_match;
     $this->user = $user;
     // Retrieve the collection from the route.
     $this->collection = $this->currentRouteMatch->getParameter('rdf_entity');
+
+    if (empty($this->collection)) {
+      if (!empty($collection_context->getRuntimeContexts(['collection'])['collection']->getContextValue())) {
+        $this->collection = $collection_context->getRuntimeContexts(['collection'])['collection']->getContextValue();
+      }
+    }
```

Comment:
I will put it but do we need to? I mean, if the block is shown to pages fulfilling the requirement of "Show on collection pages"  which checks if the page is a collection and the context exists for collections, do we need to make this check?